docs/_pages/upgradingtov7.md

```diff
@@ -10,3 +10,142 @@ sidebar:
+}
+```
+
+So all of the methods to build an assertion that used to live on the `AssertionScope` (which is what `Execute.Assertion` returned), have now moved to `AssertionChain`. This is great because it allows the second assertion to get access to the state of the first assertion. For instance, if the first assertion failed, any successive attempts to call `FailWith` will not do anything. 
+
+## No more `ClearExpectation`
+
+If you wanted to reuse the first part of the failure message across multiple failures, you could use the following construct (example taken from `TimeOnlyAssertions.BeCloseTo`):
+
+```csharp
+Execute.Assertion
+    .BecauseOf(because, becauseArgs)
+    .WithExpectation("Expected {context:the time} to be within {0} from {1}{reason}, ", precision, nearbyTime)
+    .ForCondition(Subject is not null)
+    .FailWith("but found <null>.")
+    .Then
+    .ForCondition(Subject?.IsCloseTo(nearbyTime, precision) == true)
+    .FailWith("but {0} was off by {1}.", Subject, difference)
+    .Then
+    .ClearExpectation();
+```
+
+When using an `using new AssertionScope()` construct to wrap multiple assertions, all assertions executed within that scope will reuse the same instance of `AssertionScope` (which is what `Execute.Assertion` returned). The problem was that you had to explicitly call `ClearExpectation` to prevent the failure message passed to `WithExpectation` to leak into the next assertion within that scope. People often forgot that. 
+
+We solved this in v7, by making `WithExpectation` use a nested construct. This is what it now looks like:
+
+```csharp
+assertionChain
+    .BecauseOf(because, becauseArgs)
+    .WithExpectation("Expected {context:the time} to be within {0} from {1}{reason}, ", precision, nearbyTime, chain => chain
+      .ForCondition(Subject is not null)
+      .FailWith("but found <null>.")
+      .Then
+      .ForCondition(Subject?.IsCloseTo(nearbyTime, precision) == true)
+      .FailWith("but {0} was off by {1}.", Subject, difference)
+    );
+```
+
+All the code nested within the `WithExpectation` will share the first part of the failure message, and there's no need to explicitly clear it anymore.
+
+## Amending caller identifiers with `WithPostfix`
+
+Imagine the following chained assertion
+
+```csharp
+var element = XElement.Parse(
+                """
+                <parent>
+                    <child />
+                    <child />
+                </parent>
+                """);
+
+
+element.Should().HaveElement("child", AtLeast.Twice()).Which.Should().HaveCount(1);
+```
+
+Prior to version 7, if the `HaveElement` assertion succeeded, but the `NotBeNull` failed, you would get the following exception:
+
+    Expected element to contain 1 item(s), but found 3: {<child />, <child />, <child />}.
+
+Now, in v7, it'll will return the following:
+
+    Expected element/child to contain 1 item(s), but found 3: {<child />, <child />, <child />}.
+
```

Comment:
Is this example correct?
1) `NotBeNull` ?
2) Is this `HaveCount(1)` checking anything? I think `.Which` throws, because there isn't a single returned matched element => actually returns two
3) Does this failure message really fits (the new one)? I think (apart from throwing when not `Single`) `Expected element/child to contain 1 item(s)...` isn't quite right? 🤔
Because when I read the failure, I assume following xml structure:

```xml
<parent>
    <child>
        <this_one_containing_item />
    </child>
    <child>
        <this_one_containing_item />
    </child>
</parent>
```

Reply:
No, because in this case, `HaveElement` returns an `AndWhichConstraint` that takes an `IEnumerable<T>`.